### PR TITLE
Contrib/poetry

### PR DESCRIPTION
**Pull request by StefanUlbrich**
Actions would need to be adjusted. We can copy the actions from MQV for instance. I recommend we add precommit. And I’d love to experiment with ruff—it’s the cool new kid on the block

### CODE REVIEW

File: docs/source/quickstart.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Quickstart
+
+```python
+python3 -m venv venv
+pip install "scipy<=1.7.3,>=0.16" "numpy<1.24,>=1.18"
```
Comment (sebastianneubauer) [inline docs/source/quickstart.md:5]: can we get rid of this?

**Review by sebastianneubauer — APPROVED** [review]

**Comment by StefanUlbrich** [comment]
@FelixWick : PR should be close to finished.

- Added pre-commit: (opt-in) quality checks/linting when running `git commit` and running the linters in the CI/CL. Unlike the recommended approach, the linters of the dev-dependencies are used (better reproducibility). If we want, we can add additional (locally installed) checks. Instead of flake8 et al. I opted for ruff .. incredibly fast and already widely adopted
- Adapted workflows to poetry and other changes
- Still needs an update on the Readme (will do if we agree to use poetry)

File: docs/source/quickstart.md (hunk 1 above)
Comment (StefanUlbrich) [reply docs/source/quickstart.md]: @sebastianneubauer  Still need to update the readme. For developing `poetry install` is enough to install regular/dev dependencies, `poetry install --with docs` lets you build the docs, `--with jupyter` gives you a jupyter lab etc. Don't forget to either activate the env with `poetry shell` or prepend your commands (e.g., `poetry run black .` )

Comment (StefanUlbrich) [reply docs/source/quickstart.md]: @sebastianneubauer added to `docs/source/install.md`

**Pull request merged**